### PR TITLE
kfctl binary push to GCS

### PR DESCRIPTION
**Pull request by swiftdiaries**
Addresses issue: https://github.com/kubeflow/kfctl/issues/105

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/106)

### CODE REVIEW

File: py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py [hunk 1]
```diff
@@ -285,6 +289,16 @@ def kfctl_deploy_kubeflow(app_path, project, use_basic_auth, use_istio, config_p
     apply_kubeflow(kfctl_path, app_path)
   return app_path
 
+def push_kfctl_to_gcs(app_path):
+  bucket = "kubernetes-jenkins"
```
Comment (jlewi) [inline py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py]: Why are you hardcoding the bucket here? How are we doing it for other artifacts?

**Comment by jlewi** [comment]
@swiftdiaries overall looks good; do you know why the tests are failing?

File: py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py (hunk 1 above)
Comment (swiftdiaries) [reply py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py]: I was following the same logic in the function to see if I can write to the bucket ok. I'll switch it out with these [two functions](https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/util.py#L845).

**Comment by swiftdiaries** [comment]
I think I'm not passing the bucket in the right format. Let me switch it out with the functions in the `kubeflow/testing` repo that should handle the parsing

**Comment by swiftdiaries** [comment]
I seem to be hitting quota issues. I can see the built kfctl binary [here](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_kfctl/106/kubeflow-kfctl-presubmit/1199055022751485957/artifacts/build_bin/). I checked it with a linux box and it seems to work fine

**Comment by jlewi** [comment]
Do we have an issue tracking the quota issues? I think @gabrielwen was working on fixing those?

File: py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py (hunk 1 above)
Comment (jlewi) [reply py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py]: nit: Maybe use the constant defined here: https://github.com/kubeflow/testing/blob/25b87a36f03c207f69abbc9e18500e9ac75647d0/py/kubeflow/testing/prow_artifacts.py#L19

**Comment by jlewi** [comment]
/hold
Please see my one nit-picky comment.

/lgtm
/approve

File: py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py (hunk 1 above)
Comment (swiftdiaries) [reply py/kubeflow/kfctl/testing/util/kfctl_go_test_utils.py]: Done

**Pull request merged**